l10n_th_withholding_tax_cert/models/account_move.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    wt_cert_ids = fields.One2many(
+        comodel_name="withholding.tax.cert",
+        inverse_name="move_id",
+        string="Withholding Tax Cert.",
+        readonly=True,
+    )
+    wt_cert_cancel = fields.Boolean(compute="_compute_wt_cert_cancel")
+
+    def _compute_wt_cert_cancel(self):
+        for record in self:
+            wt_certs = record.wt_cert_ids.filtered(lambda l: l.state != "cancel")
+            record.wt_cert_cancel = not wt_certs and True or False
```

Comment:
พยายามอ่านแล้ว แล้วเหมือนกับ
- wt_cert_cancel = False จะหมายถึง document ที่มี WHT Cert ที่ปกติ (ไม่ cancelled)? 
- wt_cert_cancel = True หมายถึง document ที่ไม่มี WHT Cert ที่ปกติ (ไม่ cancelled)?

ดูเหมือนมันจะเป็น no ซ้อน no ทั้งๆที่มัน yes ได้ตรงๆรึเปล่า สรุปคือ อ่านแล้วงงครับ อยากให้แก้ให้เข้าใจง่ายกว่านี้

Reply:
For statement like this,
`not wt_certs and True or False`
better change to,
`True if not wt_certs else False` or even better with `False if wt_certs else True`

Because you won't need to worry about True must be before False.